### ISSUE TITLE
sql: can't create table, index and foreign key in the same (slow) transaction

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#25663]
I was trying to QA the fix to #24626 and I ran into the following problem. I am not sure whether it is still the same issue or a new issue.

Using `cockroach sql`, with some test database.

If I copy-paste the following text in `cockroach sql` in one go, then it works without problem:

```sql
BEGIN; CREATE TABLE foo(x INT); CREATE INDEX blix ON foo(x); COMMIT;
```

It even works properly (yay!) if a table with the same name was dropped before:

```sql
CREATE TABLE foo(z INT); DROP TABLE foo;
BEGIN; CREATE TABLE foo(x INT); CREATE INDEX blix ON foo(x); COMMIT;
```

Now the problem occurs when I do the following:

1. type `CREATE TABLE foo(z INT); DROP TABLE foo;` then enter, to execute just those two statements
2. type `BEGIN;`, then *enter two times* to ensure the BEGIN statement is sent to the server. the prompt becomes ` OPEN>`
3. manually (i.e. slowly, not via copy paste), type `CREATE TABLE foo(x INT); ` then enter
4. manually (not via copy paste), type `CREATE INDEX blix ON foo(x);` then enter
5. finally type `COMMIT;` then enter

This reliably fails the transaction with 

```
pq: restart transaction: HandledRetryableTxnError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE): "sql txn" id=5088b564 key=/Table/SystemConfigSpan/Start rw=true pri=0.02325890 iso=SERIALIZABLE stat=PENDING epo=0 ts=1526655082.454945684,1 orig=1526655080.127473245,0 max=1526655080.127473245,0 wto=false rop=false seq=32
```

Even though there is zero other load on the database (I can't see why it retries!)

**Comment by knz** [comment]
@vivekmenezes can you have a look?

**Comment by vivekmenezes** [comment]
Thanks for QA-ing the fix for #24626 . This issue is the same as https://github.com/cockroachdb/cockroach/issues/21800
